migrator/migrations/m_196_to_m_197_set_poduid_where_null/migration_impl.go

```diff
@@ -0,0 +1,241 @@
+		return err
+	}
+
+	podUIDMap, err := getProcessIndicatorPodUIDMap(ctx, processIndicatorStore, processIndicatorIds)
+	if err != nil {
+		return err
+	}
+
+	err = setPodUidsUsingProcessIndicators(ctx, plopStore, podUIDMap, batchSize)
+
+	return err
+}
+
+// Returns a string which is a combination of the pod name and deploymentid so that listening_endpoints
+// can be matched to pods.
+func getPodKey(podName, deploymentID string) string {
+	// The _ character cannot appear in a podName, so it is a good separator
+	return fmt.Sprintf("%s_%s", podName, deploymentID)
+}
+
+// Get a map where the key is a combination of the pod name and deploymenid and the value is the
+// PodUid. This is later used to set the PodUids in the listening endpoints table.
+func getPodUIDMap(ctx context.Context, podStore podDatastore.Store) (map[string]string, error) {
+	podUIDMap := make(map[string]string)
+
+	err := podStore.Walk(ctx,
+		func(pod *storage.Pod) error {
+			podKey := getPodKey(pod.GetName(), pod.GetDeploymentId())
+			podUIDMap[podKey] = pod.GetId()
+			return nil
+		})
+
+	return podUIDMap, err
+}
+
+// Walks through the listening_endpoints table and sets the value of PodUid using a map obtained from the
+// pods table where the key is a combination of the pod name and deploymentid.
+func setPodUidsUsingPods(ctx context.Context, plopStore plopDatastore.Store, podUIDMap map[string]string, batchSize int) error {
+	plops := make([]*storage.ProcessListeningOnPortStorage, batchSize)
+	count := 0
+	err := plopStore.Walk(ctx,
+		func(plop *storage.ProcessListeningOnPortStorage) error {
+			// Don't set the PodUid if it is already set. The process information has the
+			// podid, so it must be set to proceed.
+			if plop.GetPodUid() == "" && plop.GetProcess() != nil {
+				podKey := getPodKey(plop.GetProcess().GetPodId(), plop.GetDeploymentId())
+				podUID, exists := podUIDMap[podKey]
+				if exists {
+					plop.PodUid = podUID
+					plops[count] = plop
+					count++
+				}
+
+				if count == batchSize {
+					err := plopStore.UpsertMany(ctx, plops)
+					count = 0
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			return nil
+		})
+
+	if count > 0 {
+		plops = plops[:count]
+		err := plopStore.UpsertMany(ctx, plops)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+// Sets PodUids by using information from the pods table. It first walks through the pods table and
+// makes a table where the key is a combination of the pod name and deploymentid and the value is the
+// PodUid. It then walks through the listening_endpoints table and uses the map to set the value of
+// PodUid where it is null and the process information exists. This is effectively a join. If the
+// process information does not exist the podid (pod name) doesn't exist, and the PodUid cannot be set
+// using this strategy.
+func setPodUIDsUsingPods(ctx context.Context, podStore podDatastore.Store, plopStore plopDatastore.Store, batchSize int) error {
```

Comment:
Similar comment to `setPodUIDsUsingProcessIndicators`.

I don't know what to name `setPodUIDsUsingPods` and `setPodUidsUsingPods`